libparsec/crates/client/src/invite/claimer.rs

```diff
@@ -470,8 +773,25 @@ impl UserClaimInProgress3Ctx {
             human_handle,
             profile,
             root_verify_key,
-        } = InviteUserConfirmation::decrypt_and_load(&payload, &self.0.shared_secret_key)
-            .map_err(ClaimInProgressError::CorruptedConfirmation)?;
+        } = match InviteUserConfirmation::decrypt_and_load(&payload, &self.0.shared_secret_key) {
+            Ok(data) => data,
+            Err(err) => {
+                let reason = match err {
+                    DataError::Decryption => CancelledGreetingAttemptReason::UndecipherablePayload,
+                    DataError::BadSerialization { .. } => {
+                        CancelledGreetingAttemptReason::UndeserializablePayload
+                    }
+                    _ => CancelledGreetingAttemptReason::InconsistentPayload,
+                };
+                if cancel_greeting_attempt(&self.0.cmds, self.0.greeting_attempt, reason)
+                    .await
+                    .is_err()
+                {
+                    // TODO: Warn about the error before discarding it
```

Comment:
```suggestion
                if let Err(e) = cancel_greeting_attempt(&self.0.cmds, self.0.greeting_attempt, reason)
                    .await
                {
                    log::warn!("error {e} while cancelling greeting attempt")
```

I'm not sure about the formatting though